drivers/counter/counter_nrfx_timer.c

```diff
@@ -115,11 +125,60 @@ static inline u32_t counter_nrfx_get_cc_value(struct device *dev,
 	return cc_val;
 }
 
+/* Function calculates distance between to values assuming that one first
+ * argument is in front and that values wrap.
+ */
+static u32_t tick_diff(u32_t new, u32_t old, u32_t top)
+{
+	if (top & (top + 1)) {
+		/* if top is not 2^n-1 */
+		u32_t diff;
+
+		diff = (new > old) ? (new - old) : new + top + 1 - old;
+
+		return diff;
+	}
+
+	return (new - old) & top;
+}
+
+static int set_cc(struct device *dev, u8_t chan_id, u32_t val)
+{
+	const struct counter_nrfx_config *nrfx_config = get_nrfx_config(dev);
+	const nrfx_timer_t *tmr = &nrfx_config->timer;
+	NRF_TIMER_Type  *reg = tmr->p_reg;
+	u32_t now;
+	u32_t top = counter_get_top_value(dev);
+	nrf_timer_task_t read_tsk =
+		   offsetof(NRF_TIMER_Type, TASKS_CAPTURE[COUNTER_READ_CC]);
+	nrf_timer_event_t evt =
+			offsetof(NRF_TIMER_Type, EVENTS_COMPARE[chan_id]);
+	u32_t int_mask = NRF_TIMER_INT_COMPARE0_MASK << chan_id;
+
+	nrf_timer_cc_write(reg, chan_id, val);
+	nrf_timer_event_clear(reg, evt);
+
+	nrf_timer_task_trigger(reg, read_tsk);
+	now = nrf_timer_cc_read(reg, COUNTER_READ_CC);
```

Comment:
Actually, this is what `nrfx_timer_capture` does, but perhaps you wanted it to be as fast as possible.